### DISCUSSION TITLE
Prolems about long mRNA reads mapping

### DISCUSSION THREAD

**Discussion by lwlive** [GoekeLab/xpore#69] [Q&A] [answered]
I have read the protocol in webpage of xpore (https://xpore.readthedocs.io/en/latest/preparation.html). It recomands us to mapping the long mRNA reads with the following commands "minimap2 -ax map-ont -uf -t 3 --secondary=no <PATH/TO/FASTQ.GZ> > <PATH/TO/SAM> 2>> <PATH/TO/SAM_LOG>".
I have read the description of the minimap2. It described like following:
-x STR preset (always applied before other options; see minimap2.1 for details) []
- map-pb/map-ont: PacBio/Nanopore vs reference mapping
- ava-pb/ava-ont: PacBio/Nanopore read overlap
- asm5/asm10/asm20: asm-to-ref mapping, for ~0.1/1/5% sequence divergence
- splice: long-read spliced alignment
- sr: genomic short-read mapping
I thought it is better to use "splice" for option "-x" in mRNA mapping. How do you thinks?
Could you give me some suggestion?

**Comment by jonathangoeke** [comment, marked as answer]
Hi @lwlive,  Nanopolish does not currently handle gapped alignments, therefore the data has to be aligned to the transcriptome (not the genome). The splice option should be used with genome alignment only. You can download transcriptome reference sequences for example from [Gencode](https://www.gencodegenes.org/human/)

**Reply by lwlive** [reply]
Hi, Mr Jonathan!  Thanks for your answer! Another two problems still confused me:
(1)Do you mean that  the direct  RNA sequencing data (fastq format) were mapped to the transcriptome reference ?
(2) I am wondering how to deal with  these  matured RNA with gap?  Are these reads not taken into consideration when predicting modidication of m6A?

**Reply by jonathangoeke** [reply]
Hi @lwlive 
(1) Yes
(2) The transcriptome reference file does not contain introns, so minimap does not need to deal with gaps in mRNAs
Only transcripts which are missing from annotations might have large gaps in the alignment, but that should not be a major issue here.

**Reply by lwlive** [reply]
OK, I got  it!  Thanks  for  your kind  reply!
